auth-service/src/main/kotlin/uk/gov/communities/delta/auth/config/ClientConfig.kt

```diff
@@ -1,20 +1,54 @@
 package uk.gov.communities.delta.auth.config
 
-data class Client(val clientId: String, val clientSecret: String)
+import org.opensaml.security.x509.BasicX509Credential
+import org.slf4j.LoggerFactory
 
-class ClientConfig(
-    private val clientSecretMarklogic: String,
-    private val clientSecretDeltaWebsite: String,
-) {
+open class Client(val clientId: String, val clientSecret: String, val samlCredential: BasicX509Credential)
+
+class OAuthClient(
+    clientId: String,
+    clientSecret: String,
+    samlCredential: BasicX509Credential,
+    val redirectUrl: String
+) : Client(clientId, clientSecret, samlCredential)
+
+class ClientConfig(val clients: List<Client>) {
     companion object {
-        fun fromEnv() = ClientConfig(
-            clientSecretMarklogic = System.getenv("CLIENT_SECRET_MARKLOGIC") ?: "dev-marklogic-client-secret",
-            clientSecretDeltaWebsite = System.getenv("CLIENT_SECRET_DELTA_WEBSITE")
-                ?: "dev-delta-website-client-secret",
-        )
+        private val logger = LoggerFactory.getLogger(Companion::class.java)
+
+        fun fromEnv(deltaConfig: DeltaConfig): ClientConfig {
+            val deltaWebsiteSecret = Env.getEnv("CLIENT_SECRET_DELTA_WEBSITE")
+            val marklogicSecret = Env.getEnvOrDevFallback("CLIENT_SECRET_MARKLOGIC", "dev-marklogic-client-secret")
+            var devDeltaWebsiteSecret = Env.getEnv("CLIENT_SECRET_DELTA_WEBSITE_DEV")
```

Comment:
So we can have the extra logic below that sets devDeltaWebsiteSecret to a fallback value only if both this and the main delta website one is null. Using `getEnvOrDevFallback` would fail on staging + prod where we want it to be null.

I don't like how I've written this tbh. I could add a `getOptionalEnvOrDevFallback` (or rename the other one to "required"). Or pass in a "DLUHC_ENV" variable and explicitly check for being in the "test" environment. Maybe that first one is slightly better?

Reply:
Wouldn't it be a bit simpler if deltaWebsiteSecret always fall back to "dev-delta-website-client-secret" (if fallback is enabled)?

There's no real reason to encourage devs to use `CLIENT_SECRET_DELTA_WEBSITE_DEV` when running it locally, is there?

And maybe a code comment explaining the purpose of devDeltaWebsiteSecret is to allow local devs to connect to the test env.

Reply:
Yeah that sounds good